OpenSim/Simulation/SimbodyEngine/CustomJoint.cpp

```diff
@@ -182,27 +182,35 @@ void CustomJoint::constructCoordinates()
         Coordinate& coord = upd_coordinates(coordIndex);
         coord.setName(coordName);
 
-
         // Determine if the MotionType of the Coordinate based
-        // on which TransformAxis it is relate to 0-2 are Rotational
+        // on which TransformAxis it operates upon: 0-2 are Rotational
         // and 3-5 are Translational. Coordinates appearing
         // in both categories are Coupled
-        Coordinate::MotionType mt = Coordinate::MotionType::Coupled;
-        for (int j = 0; j < 3; ++j){
-            if (spatialTransform[j]
-                    .getCoordinateNamesInArray().findIndex(coordName) >= 0) {
+        Coordinate::MotionType mt = Coordinate::MotionType::Undefined;
+        for (int j = 0; j < 6; ++j){
+            auto coordNamesForAxis = spatialTransform[j]
+                .getCoordinateNamesInArray();
+            if (coordNamesForAxis.findIndex(coordName) >= 0) {
                 const LinearFunction* lf = nullptr;
                 if (spatialTransform[j].hasFunction()) {
-                    lf = dynamic_cast<const LinearFunction*>(&spatialTransform[j].get_function());
-                    // displacement on axis is directly proportional to the coordinate value
+                    lf = dynamic_cast<const LinearFunction*>(
+                            &spatialTransform[j].get_function() );
+                    // if displacement on axis is directly proportional to 
+                    // the coordinate value, we have a pure rotation/translation
                     if (lf && lf->getSlope() == 1.0) {
-                        // coordinate is the axis displacement
-                        if (i < 3)
-                            // coordinate dof is about rotational axis 
+                        // coordinate is pure axis displacement
```

Comment:
According to this code, a `LinearFunction` is treated as a "pure" axis displacement regardless of the value of the intercept. Is there a test case to ensure the non-zero intercept is still being handled correctly and is not being ignored because of the "pure" MotionType? That is, does MotionType need to remain `Coupled` if the intercept is non-zero to trigger the correct code branch somewhere, regardless of the slope?